### PR TITLE
update user urls images - github public repository, add new users + v…

### PR DESCRIPTION
**Pull request by Bozhokmaria**
…alid email to check notification, getting info on qrcode page only if user have this discount

### CODE REVIEW

File: src/main/java/com/exadel/discountwebapp/userdiscount/qrcodegenerator/QRCodeService.java [hunk 1]
```diff
@@ -14,26 +16,32 @@
...
+
+            qrcodeResponse.setDiscountTitle(discount.getTitle());
+            qrcodeResponse.setPromocode(discount.getPromocode());
+            qrcodeResponse.setDiscountStartDate(discount.getStartDate());
+            qrcodeResponse.setDiscountExpirationDate(discount.getExpirationDate());
+
+            return qrcodeResponse;
+        } else {
+            throw new EntityNotFoundException(UserDiscount.class, "id", userDiscountId);
+        }
```
Comment (dmitriychernov7) [inline src/main/java/com/exadel/discountwebapp/userdiscount/qrcodegenerator/QRCodeService.java]: it is better to first check if userDiscount does not exist, and then throw exception

and then create response
in this way we will avoid "else" part

it is easier to read code in this way. 
firstly we validate data, and then process it

Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/userdiscount/qrcodegenerator/QRCodeService.java]: changed

File: src/main/java/com/exadel/discountwebapp/userdiscount/qrcodegenerator/QRCodeService.java [hunk 2]
```diff
@@ -26,14 +33,14 @@ public QRCodeResponseVO qrCodeViewPage(Long userId, Long discountId) {
...
-        qrcodeResponse.setPromocode(discount.getPromocode());
-        qrcodeResponse.setDiscountStartDate(discount.getStartDate());
-        qrcodeResponse.setDiscountExpirationDate(discount.getExpirationDate());
+            qrcodeResponse.setDiscountTitle(discount.getTitle());
+            qrcodeResponse.setPromocode(discount.getPromocode());
+            qrcodeResponse.setDiscountStartDate(discount.getStartDate());
+            qrcodeResponse.setDiscountExpirationDate(discount.getExpirationDate());
 
-        return qrcodeResponse;
+            return qrcodeResponse;
```
Comment (dmitriychernov7) [inline src/main/java/com/exadel/discountwebapp/userdiscount/qrcodegenerator/QRCodeService.java]: extra spaces are left

Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/userdiscount/qrcodegenerator/QRCodeService.java]: - extra spaces

**Review by dmitriychernov7 — APPROVED** [review]

**Review by jioter — APPROVED** [review]

**Pull request merged**